homeassistant/components/tvoverlay/notify.py

```diff
@@ -0,0 +1,270 @@
+        """Initialize the service."""
+        self.notify = Notifications(config[CONF_HOST])
+        self.is_allowed_path = is_allowed_path
+
+    async def _is_valid_url(self, url: str) -> bool:
+        """Check if a valid url and in allowlist_external_urls."""
+        if url.startswith(("http://", "https://")):
+            if not self.hass.config.is_allowed_external_url(url):
+                _LOGGER.warning(
+                    "URL is not allowed: %s, check allowlist_external_urls in configuration.yaml",
+                    url,
+                )
+                return False
+        else:
+            _LOGGER.warning("Invalid URL: %s", url)
+            return False
+        return True
+
+    async def _is_valid_file(self, filename: str) -> bool:
+        """Check if a file exists on disk and is in allowlist_external_dirs."""
+        if not self.hass.config.is_allowed_path(filename):
+            _LOGGER.warning(
+                "Path is not allowed: %s, check allowlist_external_dirs in configuration.yaml",
+                filename,
+            )
+            return False
+        if not os.path.isfile(filename):
+            _LOGGER.warning("Not a valid file: %s", filename)
+            return False
+        return True
+
+    async def _validate_image(self, image_data: str) -> str | None:
+        """Validate image_data is valid and in allowed list."""
+        if image_data.startswith(("http://", "https://")):
+            image = image_data if await self._is_valid_url(image_data) else None
+        elif image_data.startswith("mdi:"):
+            image = image_data
+        else:
+            image = image_data if await self._is_valid_file(image_data) else None
+        return image
+
+    async def _convert_to_seconds(self, duration: str | Any) -> Optional[int]:
+        """Convert string formatted duration 1w2d3h4m5s in to seconds."""
+        if not duration:
+            return None
+        if isinstance(duration, int):
+            return duration
+        duration = duration.replace(" ", "")
+        UNITS = {
+            "s": "seconds",
+            "m": "minutes",
+            "h": "hours",
+            "d": "days",
+            "w": "weeks",
+        }
+        try:
+            return int(
+                timedelta(
+                    **{
+                        UNITS.get(m.group("unit").lower(), "seconds"): float(
+                            m.group("val")
+                        )
+                        for m in re.finditer(
+                            r"(?P<val>\d+(\.\d+)?)(?P<unit>[smhdw])",
+                            duration,
+                            flags=re.I,
+                        )
+                    }
+                ).total_seconds()
+            )
+        except Exception as ex:  # pylint: disable=broad-except
+            _LOGGER.warning("Invalid duration: %s. %s", duration, ex)
+            return None
```

Comment:
Can we use a built in function? HA may have a function for this, but I'm not 100% sure.

Reply:
In fact, i wanted to use any built-in helper functions. however, I couldn't find any matching function as per my requirement.
I have asked about in discord dev channel as well: https://discord.com/channels/330944238910963714/554842238073700352/1166003572194934925
https://discord.com/channels/330944238910963714/554842238073700352/1166080899876659211

Let me know how do we proceed on this.